index.html

```diff
@@ -9,7 +9,40 @@
   </head>
   <body id="body">
     <table id = "circleGrid">
-        <!-- Generating Grid with circle using Javascirpt -->
+        <tr>
+            <td>
+                <div onmouseenter="mouseenter(event)" onmouseleave="mouseleave(event)" id= "circle0" class = "circle"></div>
+            </td>
+            <td>
+                <div onmouseenter="mouseenter(event)" onmouseleave="mouseleave(event)" id= "circle1" class = "circle"></div>
+            </td>
+            <td>
+                <div onmouseenter="mouseenter(event)" onmouseleave="mouseleave(event)" id= "circle2" class = "circle"></div>
+            </td>
+        </tr>
+
+        <tr>
+            <td>
+                <div onmouseenter="mouseenter(event)" onmouseleave="mouseleave(event)" id= "circle3" class = "circle"></div>
+            </td>
+            <td>
+                <div onmouseenter="mouseenter(event)" onmouseleave="mouseleave(event)" id= "circle4" class = "circle"></div>
+            </td>
+            <td>
+                <div onmouseenter="mouseenter(event)" onmouseleave="mouseleave(event)" id= "circle5" class = "circle"></div>
+            </td>
+        </tr>
+        <tr>
```

Comment:
you can use event bubbling to capture event and take action, rather than adding the event handlers to each of the element